### PR TITLE
v1 - [Bug] Fix issues when validation and test split modes set to 'none'

### PR DESCRIPTION
**Pull request by willyfh**
## 📝 Description

These changes follow the logic in the current main branch, where no issues occur when the split modes are set to 'none':

- To skip the validation stage, set `config.trainer.limit_val_batches` to `0.0` when `val_split_mode` is 'NONE' in the `config.py` file.
- To skip the test stage after training, include a conditional check `datamodule.test_split_mode == TestSplitMode.NONE` in the `engine.py` file.

**Note**: I'm not quite sure why those changes were removed in v1, but without those logics, the skipping process is not working properly.

🛠️ Fixes # (issue number)

## ✨ Changes

Select what type of change your PR is:

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔨 Refactor (non-breaking change which refactors the code base)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

### CODE REVIEW

File: src/anomalib/engine/engine.py [hunk 1]
```diff
@@ -476,7 +480,11 @@ def train(
         self._setup_trainer(model)
         self._setup_dataset_task(train_dataloaders, val_dataloaders, test_dataloaders, datamodule)
         self.trainer.fit(model, train_dataloaders, val_dataloaders, datamodule, ckpt_path)
-        self.trainer.test(model, test_dataloaders, ckpt_path=ckpt_path, datamodule=datamodule)
+
+        if datamodule is not None and datamodule.test_split_mode == TestSplitMode.NONE:
```
Comment (samet-akcay) [inline src/anomalib/engine/engine.py:484]: Would there be an edge case where test split is set to None, but test set is not empty? @djdameln ?
Maybe we could also check the length of the test set to be on the safe side?

**Review by samet-akcay — COMMENTED** [review]
Thanks @willyfh! I only have a single comment.

File: src/anomalib/engine/engine.py (hunk 1 above)
Comment (willyfh) [reply src/anomalib/engine/engine.py:484]: Do you mean something like this? If someone intentionally want to skip the test stage by setting the test split mode to None, but the test set is not empty, we give them a warning.

```python
if datamodule is not None and datamodule.test_split_mode == TestSplitMode.NONE:
    logger.info(f"The test_split_mode is set to '{TestSplitMode.NONE}'. Skipping test stage.")
    if len(datamodule.test_data) > 0:
        logger.warning(f"Found {len(datamodule.test_data)} images in the test set.")
    return None
```
